ui/app/AppLayouts/Wallet/views/walletconnect/WalletConnectSDK.qml

```diff
@@ -176,32 +210,117 @@ WebView {
         property bool sdkReady: false
 
+        property ListModel pairingsModel: pairings
+
+        onSdkReadyChanged: {
+            if (sdkReady) {
+                d.getPairings()
+            }
+        }
+
+        function populatePairingsModel(pairList) {
+            pairings.clear();
+            for (let i = 0; i < pairList.length; i++) {
+                pairings.append({
+                    active: pairList[i].active,
+                    topic: pairList[i].topic,
+                    expiry: pairList[i].expiry
+                });
+            }
+        }
+
+
         function isWaitingForSdk() {
             return timer.running
         }
 
-
         function generateSdkCall(methodName, paramsStr, successState, errorState) {
-            return "wcResult = {error: null}; try { wc." + methodName  + "(" + paramsStr + ").then((callRes) => { wcResult = {state: " + successState + ", error: null, result: callRes}; }).catch((error) => { wcResult = {state: " + errorState + ", error: error}; }); } catch (e) { wcResult = {state: " + errorState + ", error: \"Exception: \" + e.message}; }; wcResult"
+            return "wcResult = {}; try { wc." + methodName  + "(" + paramsStr + ").then((callRes) => { wcResult = {state: " + successState + ", error: null, result: callRes}; }).catch((error) => { wcResult = {state: " + errorState + ", error: error}; }); } catch (e) { wcResult = {state: " + errorState + ", error: \"Exception: \" + e.message}; }; wcResult"
+        }
+        function requestSdkAsync(jsCode) {
+            root.runJavaScript(jsCode,
+                function(result) {
+                    timer.restart()
+                }
+            )
         }
-        function requestSdk(jsCode) {
+
+        function requestSdk(methodName, paramsStr, successState, errorState) {
+            const jsCode = "wcResult = {}; try { const callRes = wc." + methodName + "(" + (paramsStr ? (paramsStr) : "") + "); wcResult = {state: " + successState + ", error: null, result: callRes}; } catch (e) { wcResult = {state: " + errorState + ", error: \"Exception: \" + e.message};  }; wcResult"
             root.runJavaScript(jsCode,
                 function(result) {
                     timer.restart()
                 }
             )
         }
 
+        function startListeningForEvents() {
+            const jsCode = "
+                try {
+                    async function processWCEvents() {
+                        window.wcEventResult = [];
+                        window.wcEventError = null
+                        window.wc.registerForSessionRequest((event) => {
+                            window.wcEventResult.push({name: 'session_request', payload: event});
+                        });
+                    }
+                    processWCEvents();
+                } catch (e) {
+                    window.wcEventError = e
+                }
+                window.wcEventError"
+
+            root.runJavaScript(jsCode,
```

Comment:
@stefandunca why is it not enough to call `jsCode` only once and then check in a loop (using a timer) for `window.wcEventResult` instead of calling `registerForSessionRequest` every time?

Reply:
The call `startListeningForEvents` which executes `registerForSessionRequest` is called only once after the SDK initializes.